### PR TITLE
Added Go project and instructions [API-404]

### PR DESCRIPTION
**Pull request by yuce**
~`go.mod` contains a tag I've created in my personal repo. I'll updated it when we release the Go client preview 1 tomorrow.~
`go.mod` is up to date with the latest Preview 1

### CODE REVIEW

**Comment by yuce** [comment]
Updated the PR with the v1 preview 1 release.

**Review by leszko — CHANGES_REQUESTED** [review]
Thanks for the PR @yuce . I feel like we almost have all clients covered!!! 🥇 

Added one inline comment, other than that, 2 things:
1. Could you add `go` to integration tests [here](https://github.com/hazelcast-guides/kubernetes/blob/master/.github/workflows/integration-tests.yml#L17)
2. I checked some scenarios and it seems that the go client prints some weird logs after a Hazelcast cluster is restarted. You can find details [here](https://hazelcast.atlassian.net/wiki/spaces/PM/pages/2849079475/Hazelcast+Clients+in+Kubernetes+TDD) (section "Go Client", at the bottom). Could you please check it?

File: doc/modules/ROOT/pages/index.adoc [hunk 1]
```diff
@@ -206,6 +215,28 @@ auto hz = hazelcast::new_client(std::move(config)).get();
...
+----
+import (
+	"log"
+
+	"github.com/hazelcast/hazelcast-go-client"
+)
+
+func main() {
+    cb := hazelcast.NewConfigBuilder()
+    cb.Cluster().SetAddrs("hz-hazelcast:5701")
```
[the remark this answers is not in the document]
Comment (yuce) [reply doc/modules/ROOT/pages/index.adoc]: Port is required in v1.0.0, but it will be possible in v1.1.0 to leave out the port or specify a port range.

**Comment by yuce** [comment]
@leszko  Thanks a lot for the review! I've added Go to integration tests and updated the code for v1.0.0. The weird log errors when the cluster is totally disconnected should be fixed, but when some of the members got disconnected, the client still tries to use cached addresses for those members and will print "Found a broken connection to..." to log. I've created this ticket to track that issue: https://github.com/hazelcast/hazelcast-go-client/issues/600

**Review by leszko — APPROVED** [review]
Thanks for all the changes and for creating the related GH Issue. LGTM 👍

**Comment by yuce** [comment]
Thanks!

**Pull request merged**